src/pages/Admin/Admin.tsx

```diff
@@ -108,30 +127,41 @@ type VenueDetailsProps = {
 };
 
 const VenueDetails: React.FC<VenueDetailsProps> = ({ venueId, roomIndex }) => {
-  const match = useRouteMatch();
-  const location = useLocation();
-  const { venues } = useKeyedSelector(
-    (state) => ({
-      venues: state.firestore.data.venues ?? {},
-    }),
-    ["venues"]
-  );
+  const { url: matchUrl } = useRouteMatch();
+  const { pathname: urlPath } = useLocation();
+
+  const venueSelector = useCallback(makeVenueSelector(venueId), [venueId]);
+  const venue = useSelector(venueSelector, shallowEqual);
```

Comment:
Very cool, completely your call on this, but I wonder if `hooks/useSelector` could use this behind the scenes, per https://react-redux.js.org/api/hooks#recipe-useshallowequalselector

Reply:
That is, use `shallowEqual` in `hooks/useSelector` if it makes sense.

Reply:
I would definitely want to think through/plan it more before I made a generalised call across all of our stuff. I'm not 100% sure if this is a good/best pattern, or if there is a better way to address the needs.